stdlib/selectors.pyi

```diff
@@ -31,49 +31,41 @@ class BaseSelector(metaclass=ABCMeta):
 
+class SelectSelector(_BaseSelectorImpl):
+    def register(self, fileobj: FileDescriptorLike, events: _EventMask, data: Any = None) -> SelectorKey: ...
+    def unregister(self, fileobj: FileDescriptorLike) -> SelectorKey: ...
+    def select(self, timeout: float | None = None) -> list[tuple[SelectorKey, _EventMask]]: ...
+
+class _PollLikeSelector(_BaseSelectorImpl):
+    def select(self, timeout: float | None = None) -> list[tuple[SelectorKey, _EventMask]]: ...
+
 if sys.platform != "win32":
-    class PollSelector(BaseSelector):
-        def register(self, fileobj: FileDescriptorLike, events: _EventMask, data: Any = None) -> SelectorKey: ...
-        def unregister(self, fileobj: FileDescriptorLike) -> SelectorKey: ...
-        def select(self, timeout: float | None = None) -> list[tuple[SelectorKey, _EventMask]]: ...
-        def get_map(self) -> Mapping[FileDescriptorLike, SelectorKey]: ...
+    class PollSelector(_PollLikeSelector): ...
 
 if sys.platform == "linux":
-    class EpollSelector(BaseSelector):
+    class EpollSelector(_PollLikeSelector):
         def fileno(self) -> int: ...
-        def register(self, fileobj: FileDescriptorLike, events: _EventMask, data: Any = None) -> SelectorKey: ...
-        def unregister(self, fileobj: FileDescriptorLike) -> SelectorKey: ...
-        def select(self, timeout: float | None = None) -> list[tuple[SelectorKey, _EventMask]]: ...
-        def get_map(self) -> Mapping[FileDescriptorLike, SelectorKey]: ...
 
-class DevpollSelector(BaseSelector):
+class DevpollSelector(_PollLikeSelector):
     def fileno(self) -> int: ...
-    def register(self, fileobj: FileDescriptorLike, events: _EventMask, data: Any = ...) -> SelectorKey: ...
-    def unregister(self, fileobj: FileDescriptorLike) -> SelectorKey: ...
-    def select(self, timeout: float | None = ...) -> list[tuple[SelectorKey, _EventMask]]: ...
-    def get_map(self) -> Mapping[FileDescriptorLike, SelectorKey]: ...
 
 if sys.platform != "win32":
-    class KqueueSelector(BaseSelector):
+    class KqueueSelector(_BaseSelectorImpl):
         def fileno(self) -> int: ...
-        def register(self, fileobj: FileDescriptorLike, events: _EventMask, data: Any = None) -> SelectorKey: ...
-        def unregister(self, fileobj: FileDescriptorLike) -> SelectorKey: ...
         def select(self, timeout: float | None = None) -> list[tuple[SelectorKey, _EventMask]]: ...
-        def get_map(self) -> Mapping[FileDescriptorLike, SelectorKey]: ...
 
 # Not a real class at runtime, it is just a conditional alias to other real selectors.
 # The runtime logic is more fine-grained than a `sys.platform` check;
 # not really expressible in the stubs
-class DefaultSelector(BaseSelector):
-    def register(self, fileobj: FileDescriptorLike, events: _EventMask, data: Any = None) -> SelectorKey: ...
-    def unregister(self, fileobj: FileDescriptorLike) -> SelectorKey: ...
+class DefaultSelector(_BaseSelectorImpl):
```

Comment:
Not the point of this MR, but I do wonder if this is really the best approach to the DefaultSelector? I would have thought that if we can't replicate the logic, it would make sense for it to be a union type like `DefaultSelector: SelectSelector | KqueueSelector | etc = ...`

I don't really know enough about this module to say what would be best, but it seems odd to create a new class where it never is a unique class at runtime

Reply:
If I remember correctly, that caused trouble with inheriting from `DefaultSelector`. You could try it (in a new PR) and see what mypy_primer says.

Reply:
That makes sense